client/packages/common/src/utils/BarcodeScannerContext.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { createContext, useMemo, FC } from 'react';
+import { PropsWithChildrenOnly } from '@common/types';
+import { BarcodeScanner } from '@capacitor-community/barcode-scanner';
+import { Capacitor } from '@capacitor/core';
+import { GlobalStyles } from '@mui/material';
+import { useNotification } from '../hooks/useNotification';
+import { useTranslation } from '@common/intl';
+import { Gs1Barcode, parseBarcode } from 'gs1-barcode-parser-mod';
+
+const SCAN_TIMEOUT_IN_MS = 5000;
+
+export interface ScanResult {
+  hasContent: boolean;
+  content?: string;
```

Comment:
whats the different of `hasContent === false` and `content === undefined`?

Reply:
yeah, copied the shape from the package which is being used. I was taking it directly, but have now extended so have created m own interface. I'll just make the test against `content === undefined`